src/Flunt.Br/Document/Phone.cs

```diff
@@ -1,10 +1,192 @@
 using System.Text.RegularExpressions;
 using Flunt.Br.Document.interfaces;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace Flunt.Br.Document
 {
     internal class Phone : IValidate
     {
-        public bool Validate(string value) => new Regex(@"^\(\d{2}\)\d{4}-\d{4}$").Match(value).Success;
+
+        public Phone()
+        {
+            this.format = null;
+        }
+        public Phone(string numberFormat)
```

Comment:
Corrigir o construtor de Phone, dessa forma:

` public Phone(string numberFormat)
        {
            var regex = new Regex(@"[0-9]+");
            var phonePattern = Regex.Replace(numberFormat, @"[0-9,?]+", m => { 
                var optionalsCount = m.Value.Split(new char[] { '?' }).Length - 1;
                var stringRegex = "";
                if(optionalsCount != 0)
                {
                    stringRegex += (m.Length - optionalsCount);
                    stringRegex += ",";
                    stringRegex += (optionalsCount + (m.Length - optionalsCount));
                }
                else
                {
                     stringRegex = m.Length.ToString();
                }
                return $@"\d{{{stringRegex}}}";
            });
            phonePattern = @"^" + Regex.Replace(phonePattern, @"(?:(?(1)(?!))(\+)|(?(2)(?!))(\()|(?(3)(?!))(\)))+", m => $@"\{m.Value}").Replace("-", @"\-").Replace(" ", @"\s") + "$";
            this.format = new Regex(phonePattern);
        }`

Assim é possivel passar a interrogação no formato.
`new Contract().IsPhone("99) 9999-9999", "(99) ?9999-9999", "phone", "Invalid phone");`
`new Contract().IsPhone("99) 99999-9999", "(99) ?9999-9999", "phone", "Invalid phone");`